thunder/executors/torchex.py

```diff
@@ -249,6 +249,24 @@ def _arange_transform(
     return arange(start=start, step=step, end=end, device=torch_device, dtype=torch_dtype)
 
 
+def _get_grad_transform(a: TensorProxy) -> TensorProxy:
+    return prims.get_grad(a)
+
+
+def _put_grad_transform(a: TensorProxy, val: TensorProxy) -> None:
+    # what a hack. _transform_for_operator_executor_execution needs that prims are executable
+    # by some executor, and if they are not, it will descend to the subsymbols.  So, we
+    # force that and recurse so that the symbol is not removed/no error is raised.
+    # This is breaking the intent of transform_for_operator_executor_execution, in that
+    # it is not executable.  But there will be postprocessing that will make it executable.
```

Comment:
Nothing bad about a good hack, but if you just want get_grad/put_grad to survive you could just special case put grad/get grad in the transform for execution (but there should not be any put_grad left) before you loop over the executors:

https://github.com/Lightning-AI/lightning-thunder/blob/5bdf5bc9f6805d19f5cbb152e24ed300cc109a28/thunder/executors/passes.py#L44-L55